test/e2e/main_test.go

```diff
@@ -119,3 +133,15 @@ func createNamespace(name string) (func(), error) {
 
 	return cleanup, nil
 }
+
+func isOpenshiftCluster(k8sClient client.Client) (bool, error) {
+	clusterVersion := &configv1.ClusterVersion{}
+	err := k8sClient.Get(context.TODO(), client.ObjectKey{Name: "version"}, clusterVersion)
+	if err == nil {
+		return true, nil
+	} else if strings.Contains(err.Error(), `no matches for kind "ClusterVersion" in version "config.openshift.io/v1"`) {
```

Comment:
Can we use https://pkg.go.dev/k8s.io/apimachinery/pkg/api/meta#IsNoMatchError to check the error type rather than looking at the error string?